### PR TITLE
feat(control_validator): improve roll back detection logic

### PR DESCRIPTION
**Pull request by mkquda**
## Description
- Fix rollback detection in `VelocityValidator` so near-zero target velocity is handled correctly. Previously `signbit(v_vel * t_vel)` failed when **t_vel** ≈ 0, so rollback while holding a stop was often missed.
- When `|t_vel|` is below the stop threshold, treat stop intent explicitly and flag rollback only for backward vehicle motion above `rolling_back_velocity`; otherwise keep the opposite-sign check.
- Add unit tests for `VelocityValidator` covering the main rollback cases (zero/non-zero target, with/without rollback), plus over-velocity and reverse-drive sanity cases.

## How was this PR tested?
- `colcon build --packages-select autoware_control_validator --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release`
- `colcon test --packages-select autoware_control_validator`

### CODE REVIEW

File: control/autoware_control_validator/src/control_validator.cpp [hunk 1]
```diff
@@ -164,10 +164,20 @@ void VelocityValidator::validate(
...
+      return std::signbit(v_vel);
+    }
+    return std::signbit(v_vel * t_vel);
+  }();
 
-  const bool is_rolling_back =
-    std::signbit(v_vel * t_vel) && std::abs(v_vel) > params_.rolling_back_velocity;
   if (!params_.hold_velocity_error_until_stop || !res.is_rolling_back || is_stopped) {
     res.is_rolling_back = is_rolling_back;
   }
```
Comment (zulfaqar-azmi-t4) [inline control/autoware_control_validator/src/control_validator.cpp:181]: The original processing is cheap, plus depending on developer, it will make the code looks a bit messy. so this suggestion is optional.

```suggestion
  const auto stopped_vel_th = 5e-2;
  const auto is_stopped = std::abs(v_vel) < stopped_vel_th;

if (!params_.hold_velocity_error_until_stop || !res.is_rolling_back || is_stopped) {
res.is_rolling_back = [&]() -> bool {
    if (is_stopped || std::abs(v_vel) < params_.rolling_back_velocity) {
      return false;
    }

    if (std::abs(t_vel) < stopped_vel_th) {
      return std::signbit(v_vel);
    }
    return std::signbit(v_vel * t_vel);
  }();
}
```

Comment (mkquda) [reply control/autoware_control_validator/src/control_validator.cpp:181]: [c986ad8](https://github.com/tier4/autoware_universe/pull/3315/commits/c986ad89f49bae4a4d72bd881ba158e3b2e05c1e)

**Review by yuki-takagi-66 — APPROVED** [review]
Image

LGTM. Here is the difference in my understanding
